jdbc/spring-cloud-starter-stream-source-jdbc/src/main/resources/META-INF/spring-configuration-metadata-whitelist.properties

```diff
@@ -1,3 +1,4 @@
 configuration-properties.classes=org.springframework.cloud.stream.app.jdbc.source.JdbcSourceProperties, \
-  org.springframework.cloud.stream.app.trigger.TriggerPropertiesMaxMessagesDefaultOne
+  org.springframework.cloud.stream.app.trigger.TriggerPropertiesMaxMessagesDefaultOne, \
+  org.springframework.boot.autoconfigure.jdbc.DataSourceProperties
```

Comment:
I found the issue.
The `spring.datasource` group is actually linked to 5 sources, one of which only is `org.springframework.boot.autoconfigure.jdbc.DataSourceProperties`.
[This line](https://github.com/spring-cloud/spring-cloud-dataflow/blob/f0fedcbae126a0f197508dab316bc85666c641fc/spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/ApplicationConfigurationMetadataResolver.java#L171) in the metadata resolver requires that all types are whitelisted (which kinda make sense), so it won't appear.

Workarounds include:
1) cite all 5 classes, 
2) change the resolver logic
In both cases, the outcome will be that many properties will actually show up, not only the ones in `DataSourceProperties`.

A better solution IMO is to whitelist the main properties we're actually interested in (like driver classname, url, etc) one by one.

I'll update the PR accordingly